File: 2024/template.py
FILENAME = "input.txt"


# returns two list each one with a column
def parse_puzzle():
    file = open(FILENAME, "r", encoding="utf-8")

    for line in file:
        pass


def solve_part_one(puzzle):
    pass


# *************************************
#              part 2
# *************************************


def solve_part_two(reports):
    pass


if __name__ == "__main__":
    puzzle = parse_puzzle()

    result1 = solve_part_one(puzzle)
    print(f"result1: {result1}")

    result2 = solve_part_two(puzzle)
    print(f"result2: {result2}")

File: 2024/template.rs
use std::fs::File;
use std::io::{BufRead, BufReader, Error};

#[derive(Default, Clone, Debug)]
struct Puzzle {
    prop: i32,
}

fn parse_puzzle() -> Result<Puzzle, Error> {
    let file = File::open("./input.txt")?;
    let reader = BufReader::new(file);

    let mut puzzle: Puzzle = Puzzle::default();

    for line in reader.lines() {
        // do something
    }

    Ok(puzzle)
}

fn solve_part_one(puzzle: &Puzzle) -> Result<i32, Error> {
    Ok(0)
}

/*
*************************
        part 2
*************************
*/

fn solve_part_two(puzzle: &Puzzle) -> Result<i32, Error> {
    Ok(0)
}

fn main() -> () {
    let puzzle = parse_puzzle().unwrap();

    let res1 = solve_part_one(&puzzle).unwrap();
    println!("result1: {}", res1);

    let res2 = solve_part_two(&puzzle).unwrap();
    println!("result2: {}", res2);
}
